kore/src/Kore/Step/Simplification/Exists.hs

```diff
@@ -236,45 +235,50 @@ makeEvaluateNoFreeVarInSubstitution
     -> (OrOfExpandedPattern level variable, SimplificationProof level)
 makeEvaluateNoFreeVarInSubstitution
     variable
-    patt@Predicated { term, predicate, substitution }
+    Predicated { term, predicate, substitution }
   =
     (MultiOr.make [simplifiedPattern], SimplificationProof)
   where
-    termHasVariable =
-        Set.member variable (freePureVariables term)
-    predicateHasVariable =
-        Set.member variable (freePureVariables $ unwrapPredicate predicate)
-    simplifiedPattern = case (termHasVariable, predicateHasVariable) of
-        (False, False) -> patt
-        (False, True) ->
-            let
-                predicate' = makeExistsPredicate variable predicate
-            in
-                Predicated
-                    { term = term
-                    , predicate = predicate'
-                    , substitution = substitution
-                    }
-        (True, False) ->
-            Predicated
-                { term = mkExists variable term
-                , predicate = predicate
-                , substitution = substitution
-                }
-        (True, True) ->
-            Predicated
+    hasVariable = Set.member variable
+    simplifiedPattern =
+        boundConfiguration `Predicated.andCondition` freeCondition
+      where
+        boundConfiguration
+          | hasVariable (Pattern.freeVariables term) =
+            (ExpandedPattern.topOf patternSort)
                 { term =
                     mkExists variable
-                        (ExpandedPattern.toMLPattern
-                            Predicated
-                                { term = term
-                                , predicate = predicate
-                                , substitution = mempty
-                                }
-                        )
-                , predicate = makeTruePredicate
-                , substitution = substitution
+                    $ Maybe.fromMaybe (mkTop patternSort)
```

Comment:
I know that this `mkTop patternSort` is not actually used, but, since it's here, locally it looks weird that we're adding an `exists` on something that might be a `top`. Fmapping the mkExists before calling `fromMaybe` might look better.